locals.tf

```diff
@@ -21,7 +21,7 @@ locals {
   page_rules = { for idx, item in var.page_rules : item["target"] => merge(
     item,
     {
-      priority = idx + 1
+      priority = length(var.page_rules) - idx
```

Comment:
This actually means, that on every add/delete in the list that value will change

Reply:
The priority will be adjusted accordingly